impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java

```diff
@@ -329,6 +358,49 @@ public ClientBuilder setOrgUrl(String baseUrl) {
         return this;
     }
 
+    @Override
+    public ClientBuilder setAuthorizationMode(String authorizationMode) {
+        if (Strings.isEmpty(authorizationMode)) {
+            throw new IllegalArgumentException("Invalid AuthorizationMode");
+        }
+        this.clientConfig.setAuthorizationMode(authorizationMode);
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setScopes(List<String> scopes) {
+        if (scopes == null || scopes.size() == 0) {
+            throw new IllegalArgumentException("Invalid Scopes");
+        }
+        this.clientConfig.setScopes(scopes);
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setKeyFilePath(String keyFilePath) {
+        if (Strings.isEmpty(keyFilePath)) {
+            throw new IllegalArgumentException("Invalid KeyFilePath");
+        }
+        this.clientConfig.setKeyFilePath(keyFilePath);
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setAlgorithm(String algorithm) {
+        if (Strings.isEmpty(algorithm)) {
+            throw new IllegalArgumentException("Invalid Algorithm");
+        }
+        this.clientConfig.setAlgorithm(algorithm);
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setClientId(String clientId) {
+        ConfigurationValidator.assertClientId(clientId);
+        this.clientConfig.setClientId(clientId);
```

Comment:
You need to validate that all required params are provided when using OAuth.

Reply:
ack. clientId will be validated here regardless of OAuth2 or not.